graphistry/validate/validate_graph.py

```diff
@@ -0,0 +1,61 @@
+def check_node_dataframe_exists(g, verbose=True):
+    if g._nodes is None:
+        if verbose:
+            print("Warning: graph was created with only edges. Skipping Node ID check if Node IDs match edge IDs. Use g2 = g.materialize_nodes() to force node df creation. Exiting.")
+        return False
+    return True
+
+
+def check_node_id_defined(g, verbose=True):
+    if g._node is None:
+        if verbose:
+            print("Invalid graph: Missing Node ID. Did you forget to specify the node ID in the .nodes() function? Exiting.")
+        return False
+    return True
+
+
+def check_nan_node_ids(g, verbose=True):
+    if g._nodes[g._node].isnull().any():
```

Comment:
compute & report `.sum()` vs just `.any()`

a more verbose mode might return the df to help save time